web/app/view/ReportController.js

```diff
@@ -208,15 +210,45 @@ Ext.define('Traccar.view.ReportController', {
                 params: {
                     id: positionIds
                 },
+                scope: this,
                 callback: function (records, operation, success) {
                     if (success) {
                         Ext.getStore('ReportRoute').add(records);
+                        if (this.singleEvent && records.length > 0) {
```

Comment:
I'm think maybe it's better to do it all reports anyway? Then we don't need `singleEvent` variable and everything is going to be simpler. What do you think?